### ISSUE TITLE
VSIX deployment fails with IOException

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#60031]
https://dev.azure.com/dnceng/public/_build/results?buildId=1650875&view=logs&j=b457b8f6-b646-55cc-ac29-2e3c30c16f8c&t=2c4beb43-7c29-527a-0c9b-d4b0339083dd

```
Unhandled Exception: System.IO.IOException: The process cannot access the file 'C:\Users\cloudtest\AppData\Local\Microsoft\VisualStudio\Packages\_Instances\3b699d2c\state.json' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path)
   at System.IO.File.OpenText(String path)
   at Microsoft.VisualStudio.Setup.Cache.FileStateManager.TryReadInstance(String instanceId, Instance& instance)
   at Microsoft.VisualStudio.Setup.Cache.InstanceRepository.GetInstance()
   at Microsoft.VisualStudio.Setup.Cache.CacheManager.InitPreviousLayoutRepositories()
   at Microsoft.VisualStudio.Setup.Engine.<Initialize>b__159_22()
   at Microsoft.VisualStudio.Setup.Services.ServiceProvider.AddService[T](T& object, Boolean keepAlive, Func`1 getter)
   at Microsoft.VisualStudio.Setup.Engine.Initialize()
   at Microsoft.VisualStudio.Setup.Engine.GetProducts()
   at Microsoft.VisualStudio.ExtensionManager.SetupEngineService.CreateEngine(IEngineHost engineHost, IServiceProvider serviceProvider, InstallFlags installFlags)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.PerformSetupEngineInstall(InstallableExtensionImpl extension, Boolean installPerMachine, Boolean isPackComponent, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, IProgress`1 progress, InstallFlags installFlags, AsyncOperation asyncOp, Version targetedVsVersion, IInstalledExtension& newExtension)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallInternal(InstallableExtensionImpl extension, InstallFlags installFlags, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, AsyncOperation asyncOp, IProgress`1 progress, Version targetedVsVersion)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.BeginInstall(IInstallableExtension installableExtension, InstallFlags installFlags, AsyncOperation asyncOp, Version targetedVsVersion)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.Install(IInstallableExtension extension, InstallFlags installFlags, Version targetedVsVersion)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionManagerService.Install(IInstallableExtension extension, Boolean perMachine)
   at VsixExpInstaller.Program.<>c__DisplayClass16_0.<Main>g__Install|7(IInstallableExtension installableExtension, <>c__DisplayClass16_4& , <>c__DisplayClass16_5& ) in /_/src/VSIXExpInstaller/Program.cs:line 337
   at VsixExpInstaller.Program.<>c__DisplayClass16_0.<Main>g__RunProgram|1() in /_/src/VSIXExpInstaller/Program.cs:line 312
   at VsixExpInstaller.Program.Main(String[] args) in /_/src/VSIXExpInstaller/Program.cs:line 193
```

**Comment by JoeRobich** [comment]
So I was taking a look at the Scouting integration pipeline, It is now running on VS 17.2 P1. I haven't seen the same issue with installing the extension as I have in our pipeline. I do however see one test failing consistently https://dev.azure.com/dnceng/public/_build/results?buildId=1647506&view=ms.vss-test-web.build-test-results-tab[](https://dev.azure.com/dnceng/public/_build/results?buildId=1647506&view=ms.vss-test-web.build-test-results-tab)

I was thinking that if we fixed up that test, then perhaps the easiest way to fix the integration tests is to move off 17.1 P4.

**Comment by JoeRobich** [comment]
A hunch, since I don't have permission to create a vm from the Helix image, is that perhaps AV is at fault here. I opened a PR to add a delay between installing extensions - https://github.com/dotnet/roslyn/pull/60042

Looks like the .Pre image we use will be updated to match the scouting queue tomorrow (March 9th). https://dev.azure.com/dnceng/internal/_wiki/wikis/DNCEng%20Services%20Wiki/140/VS2022-Upgrade-Schedule Likely better at this point to fix or skip the failing test mentioned above.
